fighthealthinsurance/clinicaltrials_tools.py

```
@@ -0,0 +1,393 @@
+        """Fetch JSON from url with timeout. Returns None on any failure."""
+        try:
+            async with async_timeout(timeout_secs):
+                async with session.get(url) as resp:
+                    if resp.status == 200:
+                        data: Dict[str, Any] = await resp.json()
+                        return data
+                    # 429/5xx is worth surfacing -- the registry rate-limits and
+                    # silent retries on the next query would compound the problem.
+                    log = logger.warning if resp.status >= 429 else logger.debug
+                    log(f"[clinicaltrials:{label}] HTTP {resp.status} for {url}")
+        except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)